other/web-js/client/src/data-api.js

```diff
@@ -0,0 +1,358 @@
+     *  email: "someone@example.com",
+     *  password: "mypassw0rd!",
+     * });
+     */
+    registerUser: async ({ email, password }) => {
+      return await this.registerUser("local-userpass", { email, password });
+    },
+
+    /**
+     * Log a user in with the email/password authentication provider.
+     * @param {object} - credentials The email and password to log in with.
+     * @param {string} - credentials.email The email address to log in with.
+     * @param {string} - credentials.password The password to log in with.
+     * @returns {Promise<void>}
+     * @example
+     * await dataApi.emailPasswordAuth.logIn({
+     *  email: "someone@example.com",
+     *  password: "mypassw0rd!",
+     * });
+     */
+    logIn: async ({ email, password }) => {
+      return await this.logIn("local-userpass", { email, password });
+    },
+  };
+
+  /**
+   * Call a specified Data API action endpoint.
+   * @param {string} action - The name of a Data API action to perform, e.g. "insertOne"
+   * @param {object} input - The request body for the action.
+   * @returns {object} - The response body for the action.
+   */
+  async action(action, input) {
+    if (!this.currentUser) {
+      throw new Error(`Must be logged in to call a Data API action endpoint`);
+    }
+    // If the current user access token is expired, try to refresh the
+    // session and get a new access token.
+    await this.client.refreshExpiredAccessToken();
+
+    const url = new URL(`action/${action}`, this.baseUrl).href;
+    const resp = await fetch(url, {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/ejson",
+        Accept: "application/ejson",
+        Authorization: `Bearer ${this.currentUser.access_token}`,
+      },
+      body: JSON.stringify(input),
+    });
+    const response = await resp.json();
+
+    if (resp.status === 200 || resp.status === 201) {
+      return response;
+    } else {
+      throw new Error(response);
+    }
+  }
+
+  /**
+   * @typedef {Object} FindOneInput
+   * @property {string} dataSource - The name of the data source to use.
+   * @property {string} database - The name of the database to use.
+   * @property {string} collection - The name of the collection to use.
+   * @property {object} [filter] - A MongoDB query filter to match documents.
+   * @property {object} [projection] - A MongoDB projection to control which fields are returned.
+   *
+   * @typedef {Object} FindOneResult
+   * @property {object | null} document - The document that was found, or null if no document matched the filter.
+   */
+
+  /**
+   * Find a single document in a collection.
+   * @param {FindOneInput} input - The request body for the action.
+   * @returns {Promise<FindOneResult>} - The response body for the action.
+   *
+   * @see https://mongodb.com/docs/atlas/app-services/data-api/generated-endpoints/#std-label-data-api-findOne
+   */
+  findOne = async (input) => {
```

Comment:
you could consolidate the syntax by defining as:

```suggestion
async findOne(input) {
```